src/components/sessionManager.js

```diff
@@ -1,60 +1,66 @@
+import { user as userAPI } from "@tryrolljs/api";
 import React from "react";
-import rollAPI from "../api";
+import { authSdk, apiClient } from "../api";
 
-export const SessionContext = React.createContext({ isLoggedIn: false });
+export const SessionContext = React.createContext({});
 
 export default function SessionManager({ children }) {
-  const [state, setState] = React.useState({
-    isLoggedIn: false,
-    user: { media: [{}] },
-  });
-
-  // if any authentication fails, update application state as necessary
-  const handleFail = () => {
-    console.log("failed to authenticate");
-    setState({ isLoggedIn: false });
-  };
+  const [user, setUser] = React.useState();
 
   React.useEffect(() => {
     // get user data with valid authToken
     // update application state as necessary
-    const handleSuccess = () => {
-      console.log("successfully authenticated");
-      rollAPI.user
-        .getMe()
-        .then((user) => {
-          console.log("got user data: ", user);
-          setState({ isLoggedIn: true, user });
-        })
-        .catch((err) => console.error(err));
+    const loadUserData = async () => {
+      try {
+        const user = await userAPI.getMe(apiClient);
+        setUser(user);
+      } catch (e) {
+        console.error(e);
+      }
     };
 
     // check for the existance of the code parameter in the url.
-    // if the code param exists, pass the url into the initializeSession function along
+    // if the code param exists, pass the url into the makeSession function along
     // with custom success and failure callbacks.
-    const handleInitializeSession = () => {
-      const urlParams = new URLSearchParams(window.location.search);
-      if (urlParams.has("code")) {
-        rollAPI.authentication.initializeSession(
-          window.location.search,
-          handleSuccess,
-          handleFail
-        );
+    const initializeNewSession = async () => {
+      try {
+        const urlParams = new URLSearchParams(window.location.search);
```

Comment:
perhaps we can provide a function `makeSessionFromUrl` so the consumer doesn't need to parse it themselves?

Reply:
It's now hidden inside the session manager package.